### DISCUSSION TITLE
AP2 ALAC

### DISCUSSION THREAD

**Discussion by iChUdY** [mikebrady/shairport-sync#1307] [Q&A]
Is there ALAC support in dev versions of AP2? How do I know from the log if it is active now?

**Comment by mikebrady** [comment]
At the moment, there are two types of AirPlay 2 streams — Realtime and Buffered. The Realtime stream is ALAC and the Buffered stream is AAC. Different apps and modes select one or the other automatically. The development version supports both.

**Reply by iChUdY** [reply]
So Apple Music from iOS doesn’t support lossless over AP2?

**Reply by mikebrady** [reply]
As things stand, Airplay 2 does support CD-quality lossless over a Realtime stream, but Buffered audio is (lossy) AAC. This may change with iOS 15.1 and macOS Monterey.

**Reply by iChUdY** [reply]
I hope so. I bought a hifiberry dac and studio speakers for every room in the house.

**Reply by iChUdY** [reply]
HomePod 15.1 update brings lossless now.

**Reply by mikebrady** [reply]
Yeah, thanks. Unfortunately, Apple has revealed nothing about how AirPlay 2 works, so nobody outside Apple knows the protocols.
